src/integration_test/java/com/example/Restaurant_Management_System_REST_API/rest/controller/ReservationControllerIntegrationTest.java

```diff
@@ -0,0 +1,250 @@
+                    assertNotNull(actualResponse);
+                    assertEquals(expected.getName(), actualResponse.getName());
+                    assertEquals(expected.getDescription(), actualResponse.getDescription());
+                    assertEquals(expected.getPeopleAmount(), actualResponse.getPeopleAmount());
+                    assertEquals(expected.getStart(), actualResponse.getStart());
+                    assertIterableEquals(expected.getTables(), actualResponse.getTables());
+                    assertEquals(expected.getCustomer(), actualResponse.getCustomer());
+                });
+        reservationRepository.deleteAll();
+    }
+
+    @Test
+    public void findById_ShouldMapAndReturnReservationDTOResponse_WhenReservationExist() {
+
+        webTestClient.get()
+                .uri("/api/reservation/find/" + reservation.getId())
+                .header(HttpHeaders.AUTHORIZATION, basicAuthStaffHeader)
+                .exchange()
+                .expectStatus().isOk()
+                .expectBody(ReservationDTO.class)
+                .consumeWith(response -> {
+                    ReservationDTO actualResponse = response.getResponseBody();
+                    assertNotNull(actualResponse);
+                    assertEquals(reservation.getName(), actualResponse.getName());
+                    assertEquals(reservation.getDescription(), actualResponse.getDescription());
+                    assertEquals(reservation.getPeopleAmount(), actualResponse.getPeopleAmount());
+                    assertEquals(reservation.getStart(), actualResponse.getStart());
+                    assertTrue(actualResponse.getTables().isEmpty());
+                    Customer actualCustomer = modelMapper.map(actualResponse.getCustomer(), Customer.class);
+                    assertEquals(reservation.getCustomer(), actualCustomer);
+                });
+        reservationRepository.deleteAll();
+    }
+
+    @Test
+    public void findAll_ShouldReturnReservationDTOResponseList_WhenReservationExist() {
+
+        reservation.setTables(new ArrayList<>()); //I set this as empty list, otherwise I got assertion failure null vs empty
+
+        List<ReservationDTO> expected = Arrays.asList(modelMapper.map(reservation, ReservationDTO.class));
+
+        webTestClient.get()
+                .uri("/api/reservation/findAll")
+                .header(HttpHeaders.AUTHORIZATION, basicAuthStaffHeader)
+                .exchange()
+                .expectStatus().isOk()
+                .expectBodyList(ReservationDTO.class)
+                .consumeWith(response -> {
+                    List<ReservationDTO> actualResponse = response.getResponseBody();
+                    assertNotNull(actualResponse);
+                    assertEquals(expected.size(), actualResponse.size());
+                    assertThat(actualResponse).containsExactlyInAnyOrderElementsOf(expected);
+                });
+        reservationRepository.deleteAll();
```

Comment:
This line repeats every now and then in every (almost all?) test methods. It could be improved by extracting the line to own method or by setting after each test method which will do exactly that :) Let me know if you need more help with that!

Reply:
okay I got it, here is my proposal:

![image](https://github.com/BartoliniAndBorderCollies/Restaurant_Management_System_REST_API/assets/126821059/f64b8cea-7a99-411b-b123-84fb12721566)

Reply:
I was talking about deleteAll method on repository :)

Reply:
ohh, I see, but this I also have changed:

![image](https://github.com/BartoliniAndBorderCollies/Restaurant_Management_System_REST_API/assets/126821059/30347f3f-8dbf-4789-92c1-271a4b257818)